otx/algorithms/detection/tasks/inference.py

```diff
@@ -226,18 +226,22 @@ def _init_recipe(self):
         if train_type not in (TrainType.SEMISUPERVISED, TrainType.INCREMENTAL):
             raise NotImplementedError(f"Train type {train_type} is not implemented yet.")
         if train_type == TrainType.SEMISUPERVISED:
-            if self._data_cfg.get("data", None) and self._data_cfg.data.get("unlabeled", None):
-                recipe = os.path.join(recipe_root, "semisl.py")
+            if self._is_training:
+                if self._data_cfg.get("data", None) and self._data_cfg.data.get("unlabeled", None):
+                    recipe = os.path.join(recipe_root, "semisl.py")
+                else:
+                    logger.warning("Cannot find unlabeled data.. convert to INCREMENTAL.")
+                    train_type = TrainType.INCREMENTAL
```

Comment:
I found some possible bug here. Even if we loaded it with template_semi.yaml, it can be load and run without passing unlabeled data. In this case, we need to convert to INCREMENTAL mode but since model is also hard-coded to semi (mean teacher / unbiased teacher), it would give error or run with (semi-model + incremental-mode). I dealt with this problem by changing it into model.py path double-checking the train type inside _init_model_cfg. Please refer to my [draft PR](https://github.com/openvinotoolkit/training_extensions/pull/1425/files#diff-38970adf7b9d362c1a674ca03fa4e639def08cbe36c4da5daaeea4f2f5bc6b54L212). If you have a better solution, let me know.

Reply:
Thanks for advices. We will make it run them by semi recipe but in a supervised learning way.

Reply:
As we agreed on one template for one model, we can solve this problem. @supersoob Could you check the changes